### ISSUE TITLE
Explosion radius of grenade hacks and metal doors (and possibly other explosives)

### ISSUE THREAD

**Issue by denemcam** [CleverRaven/Cataclysm-DDA#26399]
**Describe the bug**  
Explosion and shrapnel is not blocked by closed metal doors, at least from grenade hacks.

**To Reproduce**  
1. Go to lab, find grenadier zombie. 
2. Wait till it'll deploy grenade hack.
3. Close the metal door and try to run.

**Expected behavior**  
Firmly sealed room should hold all the pellets in, or at least damage the door and only after that go through it. Metal doors should be impassable for explosion wave.

**Versions and configuration(please complete the following information):**
 - OS: [Win 7]
 - Game Version [0.C-33469-g72b502c]
 - Graphics version [Tiles - default, ChestHole]
 - Mods loaded ["dda","no_npc_food","filthy_morale","novitamins","craftgp","safeautodoc","brightnights","Tolerate_This","FIC_Weapons","realguns","ew_pack","makeshift","Medieval_Stuff","More_Survival_Tools","nw_pack","DinoMod","modular_turrets","Salvaged_Robots","alt_map_key","crazy_cataclysm","mutant_npcs","npc_traits","national_guard_camp","more_locations","FujiStruct","mapgen_demo","Experimental-Z-Level","Urban_Development","boats","deoxymod","blazemod","Tanks","necromancy","more_classes_scenarios","RL_Classes","manualbionicinstall","StatsThroughSkills"]

**Additional context**  
I did not tried any other explosives, so the only thing that i saw is the grenade hack pellets and wave go through the closed metal door without damaging it.

**Comment by denemcam** [comment]
Duplicate of #25443. Just found that one.

**Comment by Leland** [comment]
Closing re: duplicate
